Tests/MeshLib/TestRasterToMesh.cpp

```diff
@@ -0,0 +1,321 @@
+    ASSERT_NEAR(0.07, bounds.second, std::numeric_limits<double>::epsilon());
+
+    std::vector<MeshLib::Node*> const& nodes = mesh->getNodes();
+    for (MeshLib::Node* n : nodes)
+        ASSERT_TRUE(((*n)[2] == 0) || ((*n)[2] == _spacing));
+
+    std::array<unsigned, 7> n_types =
+        MeshLib::MeshInformation::getNumberOfElementTypes(*mesh.get());
+    ASSERT_EQ(2 * _n_pix, n_types[6]);
+}
+
+TEST_F(RasterToMeshTest, convertRasterToHexMeshValue)
+{
+    std::unique_ptr<MeshLib::Mesh> mesh(MeshLib::RasterToMesh::convert(
+        *_raster.get(), MeshLib::MeshElemType::HEXAHEDRON,
+        MeshLib::UseIntensityAs::MATERIALS, "MaterialIDs"));
+    ASSERT_TRUE(mesh != nullptr);
+
+    ASSERT_EQ(2 * _n_nodes, mesh->getNumberOfBaseNodes());
+
+    std::vector<std::string> names =
+        mesh->getProperties().getPropertyVectorNames();
+    ASSERT_EQ(1, names.size());
+
+    MeshLib::PropertyVector<int>* prop =
+        mesh->getProperties().getPropertyVector<int>("MaterialIDs");
+    ASSERT_EQ(_n_pix, prop->size());
+
+    std::pair<int, int> const& bounds =
+        MeshLib::MeshInformation::getValueBounds<int>(*mesh.get(),
+                                                      "MaterialIDs");
+    ASSERT_NEAR(0, bounds.first, std::numeric_limits<double>::epsilon());
+    ASSERT_NEAR(0, bounds.second, std::numeric_limits<double>::epsilon());
+
+    std::vector<MeshLib::Node*> const& nodes = mesh->getNodes();
+    for (MeshLib::Node* n : nodes)
+        ASSERT_TRUE(((*n)[2] == 0) || ((*n)[2] == _spacing));
+
+    std::array<unsigned, 7> n_types =
+        MeshLib::MeshInformation::getNumberOfElementTypes(*mesh.get());
+    ASSERT_EQ(_n_pix, n_types[4]);
+}
+
+TEST_F(RasterToMeshTest, convertRasterToQuadMeshNone)
+{
+    std::unique_ptr<MeshLib::Mesh> mesh(MeshLib::RasterToMesh::convert(
+        *_raster.get(), MeshLib::MeshElemType::QUAD,
+        MeshLib::UseIntensityAs::NONE, "test"));
+    ASSERT_TRUE(mesh != nullptr);
+
+    ASSERT_EQ(_n_nodes, mesh->getNumberOfBaseNodes());
+
+    std::vector<std::string> names =
+        mesh->getProperties().getPropertyVectorNames();
+    ASSERT_TRUE(names.empty());
+
+    std::vector<MeshLib::Node*> const& nodes = mesh->getNodes();
+    for (MeshLib::Node* n : nodes)
+        ASSERT_TRUE((*n)[2] == 0);
+
+    std::array<unsigned, 7> n_types =
+        MeshLib::MeshInformation::getNumberOfElementTypes(*mesh.get());
+    ASSERT_EQ(_n_pix, n_types[2]);
+}
+
+#ifdef OGS_BUILD_GUI
+TEST_F(RasterToMeshTest, vtkImage)
+{
+    /*
+    vtkSmartPointer<VtkGeoImageSource> raster = VtkGeoImageSource::New();
+    bool read_ok = raster->readImage(QString::fromStdString(_file_name));
+    ASSERT_TRUE(read_ok);
+
+    double origin[3];
+    raster->GetOutput()->GetOrigin(origin);
+    double spacing[3];
+    raster->GetOutput()->GetSpacing(spacing);
+    GeoLib::RasterHeader const& header = _raster->getHeader();
+    raster->GetOutput()->SetDimensions(header.n_rows, header.n_cols,
+    header.n_depth);
+    */
```

Comment:
Please remove the comment block unless needed for debugging, but then add some info...